apps/common/mixins.py

```diff
@@ -0,0 +1,67 @@
+import mimetypes
+import typing
+
+from django.template.defaultfilters import filesizeformat
+from django.utils.translation import gettext
+from rest_framework import serializers
+
+from apps.project.models import ProjectAsset
+from apps.tutorial.models import TutorialAsset
+
+if typing.TYPE_CHECKING:
+    from django.core.files.base import ContentFile
+
+AssetModel = TutorialAsset | ProjectAsset
+
+
+class CommonAssetMixin:
+    if typing.TYPE_CHECKING:
+
+        class Meta:
+            model: type[AssetModel]
+
+        def validate(self, attrs: dict[str, typing.Any]) -> dict[str, typing.Any]: ...
+
```

Comment:
Do we need the validate function definition inside typing.TYPE_CHECKING?
@thenav56